### ISSUE TITLE
ActionDispatch::IllegalStateError: header already sent

### ISSUE THREAD

**Issue by vmcilwain** [vmcilwain/cnr#1]
@lucasmazza sorry for sending a message this way. I didn't see anyway to do send you a direct one on your profile. It appears that I am still running into the same bug that was supposedly resolved in ticket https://github.com/rails/rails/issues/25269. I am looking to you to determine if I should re-open that ticket or open up a ticket on the devise repo.

Please let me know

```
Failure/Error: let(:action) {get :index}
     
     ActionDispatch::IllegalStateError:
       header already sent
     Shared Example Group: "requires sign_in" called from ./spec/controllers/checklist_items_controllers_spec.rb:21
     # ./spec/controllers/checklist_items_controllers_spec.rb:22:in `block (4 levels) in <top (required)>'
     # ./spec/support/helpers/shared_examples.rb:26:in `block (2 levels) in <top (required)>'
```
This error pops up when attempting to use an rspec shared example in a controller test.
https://github.com/vmcilwain/cnr/blob/feature/checklist/spec/controllers/checklist_items_controllers_spec.rb - Line 21
https://github.com/vmcilwain/cnr/blob/feature/checklist/spec/support/helpers/shared_examples.rb - Line 23

**Comment by lucasmazza** [comment]
@vmcilwain your app is using Devise `4.1.1`, and the fix on Devise for this was released on `4.2.0`. Please update your app and see how it goes.

**Comment by vmcilwain** [comment]
@lucasmazza thanks for the feedback. I did a `gem update devise` to update devise to 4.2.0 prior to sending the message so that I wouldn't waste your time. I guess it didn't take effect. I changed my gem file to `gem 'devise', '~> 4.2.0'` and re ran the spec and things worked just fine.

Thanks for getting back to me.
